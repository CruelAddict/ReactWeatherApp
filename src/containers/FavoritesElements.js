import React from 'react'
import FavoritesElementsComponent from '../components/FavoritesElementsComponent'

export default class FavoritesElements extends React.Component {
    render() {
        return <FavoritesElementsComponent/>;
    }
}
