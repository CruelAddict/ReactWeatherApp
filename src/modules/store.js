import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import reducer from './rootReducer'
import store from "../index";

const savedItems = localStorage.getItem('favorites') ? JSON.parse(localStorage.getItem('favorites')).map(value => {
        return {
            ...value,
            isLoading: true
        };
    }) : [];

const initialState = {
    favorites: {
        items: savedItems
    },
    geo: {
        permissionGranted: null,
        coordinates: null
    }
};

export default () => {
    return createStore(
        reducer,
        initialState,
        applyMiddleware(thunk)
    )
}
