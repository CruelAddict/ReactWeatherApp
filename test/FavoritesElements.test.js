import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as Enzyme from "enzyme";
Enzyme.configure({ adapter: new Adapter() });

import FavoritesElements from '../src/containers/FavoritesElements';
import store from '../src/modules/store'
import Actions from "../src/modules/actions";



describe ('FavoritesElements', () => {

    it('Renders if store is empty', () => {
        const panel = shallow(
            <FavoritesElements
                store={store}
            />
        );
        expect(panel).toMatchSnapshot()
    });

    it('Renders if store has items', () => {
        store.dispatch(Actions.createFavorite("London")).then( () => {
            const panel = shallow(
                <FavoritesElements
                    store={store}
                />
            );
            expect(panel).toMatchSnapshot()
        });
    });

});
