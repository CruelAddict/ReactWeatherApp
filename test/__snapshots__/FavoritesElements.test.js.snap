// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesElements Renders if store has items 1`] = `<Connect(FavoritesElementComponent) />`;

exports[`FavoritesElements Renders if store is empty 1`] = `<Connect(FavoritesElementComponent) />`;
